### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by zachg99** [buzz/mediainfo.js#136]
### Issue Summary
​Video files with a large number of tracks/streams cause mediainfo.js to error out with the message: RangeError: Maximum call stack size exceeded

I'm not 100% sure what the cutoff is but most of these video files have 30+ streams, often times they are subtitles, but it seems to overrun some memory limit.

Here are the actual logging lines from the tool I use, Tdarr, that uses mediainfo.js.

```
[2023-07-25T18:00:19.070] [ERROR] Tdarr_Server - Error running MediaInfo 1
[2023-07-25T18:00:19.074] [ERROR] Tdarr_Server - RangeError: Maximum call stack size exceeded
at x (<anonymous>:wasm-function[381]:0x15c4d)
at <anonymous>:wasm-function[46]:0x5dc0
at <anonymous>:wasm-function[652]:0x21cb9
at <anonymous>:wasm-function[1023]:0x47018
at <anonymous>:wasm-function[853]:0x37827
at <anonymous>:wasm-function[3684]:0xf4884
at <anonymous>:wasm-function[3516]:0xeb5b7
at <anonymous>:wasm-function[1061]:0x487c9
at <anonymous>:wasm-function[795]:0x3006d
at <anonymous>:wasm-function[3628]:0xf01cc
[2023-07-25T18:00:19.077] [ERROR] Tdarr_Server - Error running MediaInfo 2
[2023-07-25T18:00:19.077] [ERROR] Tdarr_Server - RangeError: Maximum call stack size exceeded
at x (<anonymous>:wasm-function[381]:0x15c4d)
at <anonymous>:wasm-function[46]:0x5dc0
at <anonymous>:wasm-function[652]:0x21cb9
at <anonymous>:wasm-function[1023]:0x47018
at <anonymous>:wasm-function[853]:0x37827
at <anonymous>:wasm-function[3684]:0xf4884
at <anonymous>:wasm-function[3516]:0xeb5b7
at <anonymous>:wasm-function[1061]:0x487c9
at <anonymous>:wasm-function[795]:0x3006d
at <anonymous>:wasm-function[3628]:0xf01cc
```

### Steps to Reproduce

Get a video file with 30-45 tracks/streams
Run mediainfo.js on the file

I'm using [Tdarr](https://github.com/HaveAGitGat/Tdarr) which uses mediainfo to get video file information from files for processing. The error above is from the Tdarr logs whenever I attempt to process a video file with a bunch of tracks/streams. If i strip out all the non-english subtitle tracks to reduce the count down to like 4-6 tracks, mediainfo.js is able to scan the file and provide info needed for further processing.

**Comment by JeromeMartinez** [comment]
@zachg99 do you have a crash also with [the upstream version](https://mediaarea.net/MediaInfoOnline)?
(checking if crash is due to @buzz tweaks or something more upstream)

**Comment by zachg99** [comment]
@JeromeMartinez the upstream web version works fine. I also found the [mediainfo.js tool](https://mediainfo.js.org/) and it works there as well. I'm not sure how to check the exact version that Tdarr is using but I'll see if I can find it in the docker container this evening when I get home. If there's a way to check the version Tdarr is using via the chrome dev tools in the browser I can try that.

**Comment by buzz** [comment]
Cannot reproduce. Please supply a test file that shows this behavior with the latest version of mediainfo.js.

**Comment by HaveAGitGat** [comment]
@zachg99 please try `haveagitgat/tdarr_acc:dev_2.00.22_2023_07_26T07_49_59z`. This could be to do with https://github.com/vercel/pkg which I’ve removed from the Docker image build process. 

Tdarr is using mediainfo.js 0.1.9

**Comment by HaveAGitGat** [comment]
Seems the issue is unrelated to that so will need a sample file if possible @zachg99

**Comment by zachg99** [comment]
Will do. I'll create a sample file like the files I'm trying to process and upload it soon, shooting for this weekend.

**Comment by zachg99** [comment]
I used a [sample mkv file](https://1drv.ms/u/s!AukUlxRVz7EEhO5lDAUE7icUSaSnAw?e=xbfOFj) and then used ffprobe to add 40 subtitle tracks. This causes the error above.

**Comment by HaveAGitGat** [comment]
Thanks, tried with mediainfo.js 0.1.9 and had same issue but 0.2.0 seems to be working so will use that.

FYI there's a scope error in the example here:

https://github.com/buzz/mediainfo.js/blob/48cdcc3a13d07fc0b2e19328cc15b4a19efea5fa/src/cli.ts#L36C13-L36C13

As `const mediainfo` is declared in subscope, the `mediainfo && mediainfo.close()` won't fire as it's referencing `let mediainfo: any`

**Comment by buzz** [comment]
> FYI there's a scope error in the example here:
> 
> https://github.com/buzz/mediainfo.js/blob/48cdcc3a13d07fc0b2e19328cc15b4a19efea5fa/src/cli.ts#L36C13-L36C13

Thanks for reporting! Fix in cd044ff82d46092eb765f2fdab7d1f8d47e824ba

**Comment by buzz** [comment]
> I used a [sample mkv file](https://1drv.ms/u/s!AukUlxRVz7EEhO5lDAUE7icUSaSnAw?e=xbfOFj) and then used ffprobe to add 40 subtitle tracks. This causes the error above.

I'm not able to reproduce the error. The file parses just fine.
If you encounter another issue, please open a new ticket. Thanks
